app/controllers/autocomplete_controller.rb

```diff
@@ -0,0 +1,28 @@
+class AutocompleteController < ApplicationController
+  def search_repos
+    query = params[:query]
+    if query.present?
+      @repos = search_github_repos(query)
+    else
+      # default: current user, last updated first
+      @repos = current_user.github_search_client.repos(nil, sort: 'updated', per_page: 10, page: 1)
+    end
+    respond_to do |format|
+      format.html { render partial: 'autocomplete/repository_suggestions', locals: { repos: @repos } }
+    end
+  end
+
+  private
+
+  def search_github_repos(query)
+    # query names only, including forked repos
+    keyword = query.gsub(%r{^[a-zA-Z0-9_-]+\/}, '') + ' in:name fork:true'
+
+    # add namespace criteria if needed
+    if query.include? '/'
+      namespace = query.split('/').first
+      keyword += " user:#{namespace}"
+    end
+    current_user.github_search_client.search_repos(keyword, per_page: 10, page: 1)[:items]
+  end
+end
```

Comment:
So I think this controller can be cleaned up a little bit, we can remove some of the logic away from the `#search_repos` action.

```ruby
class AutocompleteController < ApplicationController
  def search_repos
    repos = search_github_repos(params[:query])

    respond_to do |format|
      format.html { render partial: 'autocomplete/repository_suggestions', locals: { repos: repos } }
    end
  end

  private

  def github_search_client
    @github_search_client ||= current_user.github_search_client
  end

  def search_github_repos(query)
    return github_search_client.repos(nil, sort: 'updated', per_page: 10, page: 1) unless query.present?

    keyword = query.gsub(%r{^[a-zA-Z0-9_-]+\/}, '') + ' in:name fork:true'

    # add namespace criteria if needed
    if query.include? '/'
      namespace = query.split('/').first
      keyword += " user:#{namespace}"
    end

    github_search_client.search_repos(keyword, per_page: 10, page: 1)[:items]
  end
end
```